diff --git a/event.c b/event.c
index 80d6bd6..b4056eb 100644
--- a/event.c
+++ b/event.c
@@ -833,8 +833,8 @@ event_handle_mappingnotify(xcb_mapping_notify_event_t *ev)
         foreach(_c, globalconf.clients)
         {
             client_t *c = *_c;
-            xcb_ungrab_key(globalconf.connection, XCB_GRAB_ANY, c->frame_window, XCB_BUTTON_MASK_ANY);
-            xwindow_grabkeys(c->frame_window, &c->keys);
+            xcb_ungrab_key(globalconf.connection, XCB_GRAB_ANY, c->window, XCB_BUTTON_MASK_ANY);
+            xwindow_grabkeys(c->window, &c->keys);
         }
     }
 }
diff --git a/objects/client.c b/objects/client.c
index 703362d..9beaf96 100644
--- a/objects/client.c
+++ b/objects/client.c
@@ -2283,7 +2283,7 @@ luaA_client_keys(lua_State *L)
     {
         luaA_key_array_set(L, 1, 2, keys);
         luaA_object_emit_signal(L, 1, "property::keys", 0);
-        xcb_ungrab_key(globalconf.connection, XCB_GRAB_ANY, c->frame_window, XCB_BUTTON_MASK_ANY);
+        xcb_ungrab_key(globalconf.connection, XCB_GRAB_ANY, c->window, XCB_BUTTON_MASK_ANY);
         xwindow_grabkeys(c->window, keys);
     }
 
